pkg/controller/network_policy.go

```diff
@@ -228,25 +228,40 @@ func (c *Controller) handleUpdateNp(key string) error {
 			svcAsName = svcAsNameIPv6
 			svcIPs = svcIpv6s
 		}
-		if err = c.ovnLegacyClient.CreateNpAddressSet(svcAsName, np.Namespace, npName, "service"); err != nil {
-			klog.Errorf("failed to create address_set %s, %v", svcAsNameIPv4, err)
+
+		if err = c.ovnClient.CreateAddressSet(svcAsName, map[string]string{
+			networkPolicyKey: fmt.Sprintf("%s/%s/%s", np.Namespace, np.Name, "service"),
+		}); err != nil {
+			klog.Errorf("create address set %s for np %s: %v", svcAsName, key, err)
 			return err
 		}
-		if err = c.ovnLegacyClient.SetAddressesToAddressSet(svcIPs, svcAsName); err != nil {
-			klog.Errorf("failed to set netpol svc, %v", err)
+
+		if err = c.ovnClient.AddressSetUpdateAddress(svcAsName, svcIPs...); err != nil {
+			klog.Errorf("set service ips to address set %s: %v", svcAsName, err)
 			return err
 		}
 	}
 
-	var ingressAclCmd []string
-	exist, err := c.ovnClient.PortGroupExists(pgName)
-	if err != nil {
-		klog.Errorf("failed to query np %s port group, %v", key, err)
+	// we should first delete acl and address_set before update
+	if err = c.ovnClient.DeleteAcls(pgName, portGroupKey, "", nil); err != nil {
+		klog.Errorf("delete np %s acls: %v", key, err)
+		return err
+	}
+
+	if err := c.ovnClient.DeleteAddressSets(map[string]string{
+		networkPolicyKey: fmt.Sprintf("%s/%s/%s", np.Namespace, np.Name, "ingress"),
+	}); err != nil {
+		klog.Errorf("delete np %s ingress address set: %v", key, err)
 		return err
 	}
-	if exist {
-		ingressAclCmd = []string{"--type=port-group", "acl-del", pgName, "to-lport"}
+
+	if err := c.ovnClient.DeleteAddressSets(map[string]string{
+		networkPolicyKey: fmt.Sprintf("%s/%s/%s", np.Namespace, np.Name, "egress"),
+	}); err != nil {
+		klog.Errorf("delete np %s egress address set: %v", key, err)
+		return err
 	}
```

Comment:
We should update the ACLs instead of creating them. To be more specific, we should update ACLs in a single transaction, just like ovn-nbctl does.